front/lib/api/assistant/generation.ts

```diff
@@ -165,10 +193,14 @@ export async function renderConversationForModel({
 
   // We initialize `tokensUsed` to the prompt tokens + a bit of buffer for message rendering
   // approximations, 64 tokens seems small enough and ample enough.
-  let tokensUsed = promptCountRes.value + 64;
+  const tokensMargin = 64;
+  let tokensUsed = promptCountRes.value + tokensMargin;
 
   // Go backward and accumulate as much as we can within allowedTokenCount.
   const selected = [];
+  const truncationMessage = `... (content truncated)`;
+  const approxTruncMsgTokenCount = truncationMessage.length / 3;
```

Comment:
Based on https://platform.openai.com/tokenizer (legacy) should be 6

Reply:
I see 4 rather than 6 in the link? Here we want an upper bound on the tokens rather than an average, that's why I used something more conservative